### DISCUSSION TITLE
How to use props of custom component in q-input

### DISCUSSION THREAD

**Discussion by yuanbing** [quasarframework/quasar#11035] [General - Components / Directives / etc] [answered]
Hi,

I'm building a application that does the following:

1. The application displays a list of orders (via q-table)
2. Once the user clicks on one of the orders, a dialog (via dialog-plugin) will pop up with one editable input field and a list of order status updates (descending sorted via the "lastUpdate" timestamp). 
3. the user can use input field to update the status of the order.
4. when the user hits the "enter" while on the input field, the order will be updated and dialog will disappear.

I've followed the instructions [here](https://quasar.dev/quasar-plugins/dialog#invoking-custom-component): I've included the custom dialog:
[order-detail.vue.txt](https://github.com/quasarframework/quasar/files/7370542/order-detail.vue.txt)

Apparently, this approach doesn't work:

1. there was error saying that I CANNOT use v-model to bind to property "order" or "order.status";
2. what I want to do here is to use the "order.status" as the default value of the input field. 
3. And when the user hits the "enter" key, I would like to emit an event with the current input to update the order status in DB. 

Can someone shed some light how I could achieve this?

Many thanks in advance!

**Comment by metalsadman** [comment, marked as answer]
1. you can't mutate a prop https://v3.vuejs.org/guide/component-props.html#one-way-data-flow. you should debug it in your setup function log the prop in question.
2. use computed prop or local value that takes your prop which you will assign as v-model of your input.
3. on your focus input inhibit `@keydown.enter` event, then call the function below from your code take note of the comments in the code you copied from the docs, you can pass a payload to the onDialogOK event, that you can consume in your `Dialog.onOk` chained callback.
```js
// other methods that we used in our vue html template;
      // these are part of our example (so not required)
      onOKClick() {
        console.log(dialogRef);
        // on OK, it is REQUIRED to
        // call onDialogOK (with optional payload)
        onDialogOK({
          status: order.status,
        });
        // or with payload: onDialogOK({ ... })
        // ...and it will also hide the dialog automatically
      },
```
